File: 0242-valid-anagram/0242-valid-anagram.py
"""
HashMap + counter
"""

from collections import Counter

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_map = Counter(list(s))
        
        for c in list(t):
            if c not in s_map:
                return False
            else:
                s_map[c] -= 1
        
        return all(num == 0 for num in list(s_map.values()))

File: 0242-valid-anagram/0242-valid-anagram.java
import java.util.*;

class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character, Integer> map = new HashMap<>();
        for (char c: s.toCharArray()) {
            map.put(c, map.getOrDefault(c, 0) + 1);
        }
        for (char c: t.toCharArray()) {
            map.put(c, map.getOrDefault(c, 0) - 1);
        }

        return map.values()
            .stream()
            .allMatch(v -> v == 0);
    }
}